htmd/parameterization/detect.py

```diff
@@ -0,0 +1,366 @@
+# (c) 2015-2017 Acellera Ltd http://www.acellera.com
+# All Rights Reserved
+# Distributed under HTMD Software License Agreement
+# No redistribution in whole or part
+#
+import logging
+from collections import OrderedDict
+import itertools
+import networkx as nx
+from periodictable import elements
+
+logger = logging.getLogger(__name__)
+
+
+def getMolecularGraph(molecule):
+    """
+    Generate a graph from the topology of molecule
+
+    The graph nodes represent atoms and the graph edges represent bonds. Also, the graph nodes store element
+    information.
+    """
+
+    graph = nx.Graph()
+    for i, element in enumerate(molecule.element):
+        graph.add_node(i, element=element, number=elements.symbol(element).number)
+    graph.add_edges_from(molecule.bonds)
+
+    return graph
+
+def getMolecularTree(graph, source):
+    """
+    Generate a tree from a molecular graph
+
+    The tree starts from source node (atom) and grows along the edges (bonds) unrolling all encountered loops.
+    The tree grows until all the nodes (atoms) of the graph are included.
+    """
+
+    assert nx.is_connected(graph)
+
+    tree = nx.DiGraph()
+    tree.add_node(0, base=source, element=graph.nodes[source]['element'])
+    current_nodes = list(tree.nodes)
+    base_nodes = {source}
+
+    while True:
+        new_nodes = []
+        neighbor_filter = lambda node: node not in base_nodes
+        for current_node in current_nodes:
+            for neighbor in filter(neighbor_filter, graph.neighbors(tree.nodes[current_node]['base'])):
+                new_node = len(tree.nodes)
+                tree.add_node(new_node, base=neighbor, element=graph.nodes[neighbor]['element'])
+                tree.add_edge(current_node, new_node)
+                new_nodes.append(new_node)
+
+        current_nodes = new_nodes
+        base_nodes = {base for _, base in tree.nodes.data('base')}
+        if base_nodes == set(graph.nodes):
+            break
+
+    return tree
+
+def checkIsomorphism(graph1, graph2):
+    """
+    Check if two molecular graphs are isomorphic based on topology (bonds) and elements.
+    """
+
+    return nx.is_isomorphic(graph1, graph2, node_match = lambda node1, node2: node1['element'] == node2['element'])
+
+def detectEquivalentAtoms(molecule):
+    """
+    Detect topologically equivalent atoms.
+
+    Arguments
+    ---------
+    molecule : FFMolecule
```

Comment:
`:class:\`FFMolecule <htmd.ffmole.blabla.FFMolecule>\`` for the online docs to work

Reply:
Done!